lib/ansible/plugins/connection/psrp.py

```diff
@@ -166,6 +166,7 @@
 from ansible.plugins.shell.powershell import _common_args
 from ansible.utils.hashing import secure_hash
 from ansible.utils.path import makedirs_safe
+from requests.exceptions import ConnectTimeout
```

Comment:
Should be a conditional import in case requests isn't installed on the controller

Reply:
If requests isn't installed, pypsrp will fail anyhow, so I guess this should go with the pypsrp imports ?

Reply:
That is true but I believe the plugin can still be loaded even when not actually being used. @bcoca is that true?

Reply:
@jborean93 If that's the case, it would fail equally well for WSMan, but that wasn't an issue before. So this shouldn't be a problem.

Reply:
I am probably wrong but I swear there was a case where we load the plugin even when it may not be used but I can't for the life of me remember what it could have been. Hopefully bcoca can let me know whether that is true or not.

So if the above is true if requests was not in the try/except block and Ansible goes to load the plugin (even when the user didn't request it) then it will bomb out.

Reply:
@jborean93 Sure, if it was *not* inside the try/except. But it is now inside the try/except together with pypsrp, so I think the current code is fine.

Reply:
Yep agreed, happy with how it is now, was just a comment with it being outside when I added the comment.